src/plugins/items/shopping/item-value.plugin.ts

```diff
@@ -1,19 +1,60 @@
 import { itemInteractionActionHandler } from '@engine/action';
-import { Shop } from '@engine/config/shop-config';
-import { widgets } from '@engine/config/config-handler';
+import { findItem, findShop, widgets } from '@engine/config/config-handler';
+import { getItemFromContainer } from '@engine/world';
 
-export const shopSellValueHandler: itemInteractionActionHandler = ({ player, itemDetails }) => {
-    const itemValue = itemDetails.value || 1;
-    player.sendMessage(`${itemDetails.name}: currently costs ${itemValue} coins.`);
+export const shopSellValueHandler: itemInteractionActionHandler = (details) => {
+    const { player, itemId, itemSlot, widgetId, option } = details;
+
+    if(!player.interfaceState.findWidget(widgetId)) {
+        return;
+    }
+
+    const openedShopKey = player.metadata.lastOpenedShopKey;
+    if(!openedShopKey) {
+        return;
+    }
+
+    const shop = findShop(openedShopKey);
+    if(!shop) {
+        return;
+    }
```

Comment:
We should probably have some error logs here to catch naughty players in future

Doesn't need to be done in this PR, probably applies to a lot of our system